tools/check-proto-compatibility/index.ts

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env node
+
+import { getFilePathsSync } from '../utils/getFilePathsSync'
+import { checkProtoComponent, ReturnValue } from './checkProtoComponent'
+
+function getParam(key: string) {
+  const index = process.argv.findIndex((item) => item === key)
+  return index !== -1 && process.argv.length > index && process.argv[index + 1]
+}
+
+async function main() {
+  const definitionsPath = getParam('--definitions-path')
+  if (!definitionsPath) {
+    console.error('Arg --definitions-path is required.')
+    process.exit(2)
+  }
+
+  const protoComponents = getFilePathsSync(definitionsPath, false)
+    .filter((filePath) => filePath.toLowerCase().endsWith('.proto'))
+    .map((filePath) => filePath.substring(0, filePath.length - 6))
+
+  const protoComponentPromisesResult = await Promise.all(
+    protoComponents.map((component) =>
+      checkProtoComponent(definitionsPath, component)
```

Comment:
```suggestion
      { returnType: checkProtoComponent(definitionsPath, component), component }
```
continue reading to understand why ☝🏼